Encryption.md

```diff
@@ -132,49 +203,164 @@ the encryption buffer is comprised of an IV, ciphertext and tag, as described in
 Algorithms section. The length of the encryption buffer (a 4-byte little endian) is 
 written to the output stream, followed by the buffer itself.
 
+|length (4 bytes) | IV (12 bytes) | ciphertext (`length` bytes) | tag (16 bytes) |
```

Comment:
+1 for including this diagram.

Why is IV between length and ciphertext? I think IV should come before length. Otherwise, it isn't clear that length is the ciphertext length and not the length of IV, ciphertext, and tag, or for CTR, just IV and ciphertext.